eclair-core/src/main/scala/fr/acinq/eclair/api/MetaService.scala

```diff
@@ -0,0 +1,9 @@
+package fr.acinq.eclair.api
+
+import akka.http.scaladsl.server.Route
+
+trait MetaService {
```

Comment:
is this really needed?

Reply:
Yes otherwise we can't access the field "route" (see Setup.scala#L310) on two objects with different type, this is _transitory_ and will be removed as soon as we deprecate the old API.

Reply:
How about something like:

```scala
case class ServiceBase(eclairApi: EclairApi, password: String, actorSystem: ActorSystem,  mat: ActorMaterializer)
```

which can then be used like so:

```scala
new ServiceBase(eclair, password, system, materializer) with NewService
```

assuming that `NewService` has been modified like this:
```scala
trait NewService extends Directives with Logging {

  this: ServiceBase =>
```